test/metabase/api/action_test.clj

```diff
@@ -203,6 +209,37 @@
                   (is (nil? (mt/user-http-request :crowberto :delete 204 action-path)))
                   (is (= "Not found." (mt/user-http-request :crowberto :get 404 action-path))))))))))))
 
+(deftest snowplow-test
+  (snowplow-test/with-fake-snowplow-collector
+    (mt/with-actions-enabled
+      (testing "Should track when"
+        (mt/with-actions [{card-id :id} {:dataset true :dataset_query (mt/mbql-query users)}]
```

Comment:
The `mt/with-actions` function has handy defaults for each action type, so creating new actions with POST requests is overkill. `list-actions-test` is probably doing something specific that it requires a special list of actions. Instead, do something like this:
```
(doseq [type [:implicit :http :query]] 
  (mt/with-actions [{:keys [action-id]} {:type type}]
```
And then you can `db/select-one` for the action data. You can look at `do-with-action` for what the defaults look like.

Reply:
Just realised this comment wasn't completely clear on it's intention. I meant to say you shouldn't need to extract and use `all-actions-default`, since reasonable defaults are maintained in the implementation of the with-actions macro

Reply:
good point, I replaced with just a simple `with-temp`

Reply:
Okay, that makes sense too